### PR TITLE
Refactor/remove detroit

### LINKED ISSUE #37: Completely remove `lives_in_detroit`
Update code to use location rather than this boolean

Change peer_group.rb to group users based on location, rather than using the detroit boolean

### PR DESCRIPTION
**Pull request by rebeccafae**
## Migrations
YES

## Description
Group peers based on location

## Github Issue
Fixes #37

## Definition of Done

- [X] This PR has appropriate test coverage.
- ~~[ ] Documentation has been updated. *(if needed)*~~

### CODE REVIEW

**Comment by rebeccafae** [comment]
@aelydens and @westonplatter can you both please review this?

File: spec/models/peer_group_spec.rb [hunk 1]
```diff
@@ -1,32 +1,55 @@
...
+      end
+
+      let(:detroit) { create :location, city: "Detroit"}
+      let!(:detroit_users) do
+        create_list(:skinny_user, 2, :groupable, location_id: detroit.id)
+      end
+
+      it "does not group denver user with detroit users" do
+        PeerGroup.generate_groups
+        @groups = PeerGroup.all.map(&:users)
```
Comment (marshallds) [inline spec/models/peer_group_spec.rb]: I don't think we actually need instance variables...

File: app/views/users/edit.html.erb [hunk 2]
```diff
@@ -131,3 +130,11 @@
     </div>
   </main>
 <%end%>
+
+<script>
```
Comment (aelydens) [inline app/views/users/edit.html.erb]: Probably worth pulling this out of the view and into a separate js file, just to encourage future contributors to follow that pattern.

File: app/views/users/edit.html.erb [hunk 3]
```diff
@@ -41,24 +41,23 @@
...
-                  <p>What if I don't live in Detroit?<p>
-                  <p>
-                    You will not be placed in a peer group and can only mentor.
-                    However, we're working hard to expand Women Rising to other areas. Stay tuned!
-                  </p>
-                </div>
+                <%= f.collection_select(:location_id, Location.all, :id, :city_state, include_blank: 'Other') %>
+              </div>
+            </h4>
+            <h4 class="medium-frame hidden" id="zip-code">Zip Code:
```
Comment (aelydens) [inline app/views/users/edit.html.erb:52]: Nice!

**Review by aelydens — APPROVED** [review]
Looks good. Thanks for updating the specs!

File: app/models/peer_group.rb [hunk 4]
```diff
@@ -26,25 +26,27 @@ def send_mail
...
 
-  def self.automatically_create_groups
+  def self.automatically_create_groups(location)
     users = User.where(
       is_participating_this_month: true,
       waitlist: false,
-      live_in_detroit: true,
-      is_assigned_peer_group: false
+      is_assigned_peer_group: false,
+      location_id: location
```
Comment (westonplatter) [inline app/models/peer_group.rb]: What value is stored in `location`, integer or an ActiveRecord instance? If it's the an integer (ie, a Location instance `id`), let's change this to `location_id`.

File: spec/features/user_information_spec.rb [hunk 5]
```diff
@@ -17,7 +17,29 @@
     expect(page).to have_content 'Location'
     select 'Boulder, CO'
     click_on 'Submit'
-    expect(user.reload.location).to_not be_nil
+    expect(user.reload.location).to eq boulder
     expect(page).to have_content user.location.city_state
   end
+
+  scenario 'can choose other location' do
```
Comment (westonplatter) [inline spec/features/user_information_spec.rb:24]: Nice work on adding specs here!

**Review by westonplatter — CHANGES_REQUESTED** [review]

**Comment by rebeccafae** [comment]
@westonplatter I addressed your comment, please re-review

**Comment by westonplatter** [comment]
@rfjackson11 sweet. looks good to me.

**Review by westonplatter — APPROVED** [review]

**Comment by westonplatter** [comment]
Great work @rfjackson11!!

**Pull request merged**